server/lib/tuist/authorization.ex

```diff
@@ -17,122 +16,183 @@ defmodule Tuist.Authorization do
-      allow [:authenticated_as_project, :projects_match]
+      allow([:authenticated_as_project, :projects_match])
     end
   end
 
   object :account_registry do
     action :read do
-      desc "Allows users of an account to read its registry."
-      allow [:authenticated_as_user, user_role: :user]
+      desc("Allows users of an account to read its registry.")
+      allow([:authenticated_as_user, user_role: :user])
 
-      desc "Allows the admin of an account to read its registry."
-      allow [:authenticated_as_user, user_role: :admin]
+      desc("Allows the admin of an account to read its registry.")
+      allow([:authenticated_as_user, user_role: :admin])
 
-      desc "Allows the authenticated account to read the account registry if it matches the account whose registry is being read."
-      allow [:authenticated_as_account, :accounts_match, scopes_permit: :account_registry_read]
+      desc(
+        "Allows the authenticated account to read the account registry if it matches the account whose registry is being read."
+      )
 
-      desc "Allows the authenticated project to read the cache if it matches the project whose cache is being read."
-      allow [:authenticated_as_project, :accounts_match]
+      allow([:authenticated_as_account, :accounts_match, scopes_permit: :account_registry_read])
+
+      desc("Allows the authenticated project to read the cache if it matches the project whose cache is being read.")
+
+      allow([:authenticated_as_project, :accounts_match])
     end
   end
 
   object :account_settings do
     action :update do
-      desc "Allows the admin of an account to update its settings."
-      allow [:authenticated_as_user, user_role: :admin]
+      desc("Allows the admin of an account to update its settings.")
+      allow([:authenticated_as_user, user_role: :admin])
     end
   end
 
   object :account_token do
     action :create do
-      desc "Allows users of an account to create an account token."
-      allow [:authenticated_as_user, user_role: :user]
+      desc("Allows users of an account to create an account token.")
+      allow([:authenticated_as_user, user_role: :user])
 
-      desc "Allows the admin of an account to create an account token."
-      allow [:authenticated_as_user, user_role: :admin]
+      desc("Allows the admin of an account to create an account token.")
+      allow([:authenticated_as_user, user_role: :admin])
     end
   end
 
   object :account_organization do
     action :update do
-      desc "Allows the admin of an account to update its organization."
-      allow [:authenticated_as_user, user_role: :admin]
+      desc("Allows the admin of an account to update its organization.")
+      allow([:authenticated_as_user, user_role: :admin])
+    end
+  end
+
+  object :project_preview do
+    action :create do
+      desc("Allows users of a project to create a preview.")
+      allow([:authenticated_as_user, user_role: :user])
```

Comment:
not a huge fan of this auto-formatting 🤔 but also don't want to rewrite these things by hand 😬